src/components/Profile/EditProfilePopup/EditProfilePopup.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect, useCallback } from 'react';
+
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import Chip from '@material-ui/core/Chip';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import TextField from '@material-ui/core/TextField';
+
+import { useAuth } from '../../../services/AuthContext';
+import { db, storage, getCurrentUser } from '../../../services/firebase';
+
+const EditProfilePopup = ({ isOpen, onDismiss }) => {
+  const { userData, updateUserData } = useAuth();
+
+  // State for list of interests
+  // TODO: Could write new interests to DB when user inputs something new?
+  const [interestsOptions, setInterestsOptions] = useState([]);
+
+  // State related to form values
+  const [profilePicFile, setProfilePicFile] = useState(null);
+  const [personalInfo, setPersonalInfo] = useState({});
+  const [contactInfo, setContactInfo] = useState({});
+  const [userInterests, setUserInterests] = useState([]);
+
+  const resetFormValues = useCallback(() => {
+    setPersonalInfo(userData.personalInfo);
+    setContactInfo(userData.contactInfo);
+    setUserInterests(userData.interests ? userData.interests : []);
+  }, [userData]);
+
+  // Every time modal dialog opens/closes, reset the form value state.
+  useEffect(() => {
+    resetFormValues();
+  }, [resetFormValues, isOpen]);
+
+  // Get list of interests from db
+  useEffect(() => {
+    db.ref('interests/').on('value', (snapshot) => {
```

Comment:
this can be a 'once' instead of an 'on', since you don't need to have it constantly be updated for the interests table.

Reply:
Sounds good to me. Thanks!